lib/license_finder/package_managers/yarn.rb

```diff
@@ -2,16 +2,20 @@
 
 module LicenseFinder
   class Yarn < PackageManager
-    SHELL_COMMAND = 'yarn licenses list --no-progress --json'
+    SHELL_COMMAND = 'yarn licenses list --json'
 
     def possible_package_paths
       [project_path.join('yarn.lock')]
     end
 
     def current_packages
+      #keeping the use of yarn1_production_flag here because the license plugin supports same commands as licenses in yarn v1
       cmd = "#{Yarn::SHELL_COMMAND}#{yarn1_production_flag}"
-      suffix = " --cwd #{project_path}" unless project_path.nil?
-      cmd += suffix unless suffix.nil?
+      unless yarn2_project?
+        cmd += " --no-progress"
+        suffix = " --cwd #{project_path}" unless project_path.nil?
+        cmd += suffix unless suffix.nil?
```

Comment:
could we refactor these two lines into one?
```ruby
cmd += " --cwd #{project_path}" unless project_path.nil?
```

Reply:
refactored